### ISSUE TITLE
`r` isn't uniformly distributed over the annulus

### ISSUE THREAD

**Issue by TuurDutoit** [TuurDutoit/bridson#1]
I got this email from someone:

> I just felt I should mention that your implementation of bridson has a bug.
>     You select the candidate from the annulus using a simple polar coordinate. 
>     Sadly, 'r' isn't uniformly distributed over the annulus. Fortunately, r^2 is.
>     Selecting with
>     `r = sqrt(random(r*r, r2*r2));`
>     will fix the bug.
>     If you plot an annulus with your existing samples, it'll have a lot of points near 'r'.
>     if you plot with the fix, the annulus is properly uniformly distributed.

Figure out what he means and fix the problem.

**Comment by fenwick67** [comment]
When you have a circle or ring that you sample in polar coordinates (that is, you pick a random angle and random radius), the points are more bunched together toward the center.

codepen example:

http://codepen.io/fenwick/pen/qqQzLm?editors=1010

![example](http://i.imgur.com/WZdBrbB.png)

This is the line you need to change: 

https://github.com/TuurDutoit/bridson/blob/master/index.js#L105

**Comment by TuurDutoit** [comment]
Yeah, that's what I thought, just haven't found the time to fix it yet. I'll do it as soon as possible.
Thanks for the example, though!

**Comment by TuurDutoit** [comment]
Implemented a fix in a new branch `better-distribution`. Can you check if that's correct?
